lightning/src/ln/channelmonitor.rs

```diff
@@ -398,44 +391,17 @@ pub(crate) const LATENCY_GRACE_PERIOD_BLOCKS: u32 = 3;
 /// keeping bumping another claim tx to solve the outpoint.
 pub(crate) const ANTI_REORG_DELAY: u32 = 6;
 
-enum Storage<ChanSigner: ChannelKeys> {
-	Local {
-		keys: ChanSigner,
-		funding_key: SecretKey,
-		revocation_base_key: SecretKey,
-		htlc_base_key: SecretKey,
-		delayed_payment_base_key: SecretKey,
-		payment_base_key: SecretKey,
-		funding_info: Option<(OutPoint, Script)>,
-		current_remote_commitment_txid: Option<Sha256dHash>,
-		prev_remote_commitment_txid: Option<Sha256dHash>,
-	},
-	Watchtower {
-		revocation_base_key: PublicKey,
-		htlc_base_key: PublicKey,
-	}
+struct OnchainDetection<ChanSigner: ChannelKeys> {
```

Comment:
I dont think "OnchainDetection" is any better of a name. This should just be flattened and all the fields become a field directly in ChannelMonitor.